WordPress/openverse-frontend
# Add descriptive names to e2e response tapes

## Fixes


Fixes #1050 by @sarayourfriend

## Description


To make the end-to-end tests less flaky, we save the API responses and re-use them in subsequent tests. This way, the rendered pages use the same data, and should render the same result. 

These responses are saved in `/test/tapes/` folder as `response-<number>.json5`. When trying to fix the end-to-end tests on #1159, I had difficulties debugging. It is unclear, which files are changed when you run e2e tests with`:update-tapes`.

The main changes in this PR are in `test/proxy.js` file, and the rest of the 300+ file changes are just the renaming and date updates.

This PR uses `requestType_mediaType_queryOrUuid_connectionHeader` pattern:
-  thumbs: `thumb_audio_4f5c37fd-e6ba-4aa8-b06a-529aed468ca4_closed.json5`
-  related: `related_images_e9d97a98-621b-4ec2-bf70-f47a74380452_keep-alive.json5`
-  search: `search_images_q=243f6a8885a308d3.json5`
-  detail responses: `detail_images_802d76cd-84cf-45bc-81ac-610331dab420.json5`

While working on this PR, I noticed had the same url, but differed in their connection headers and compression. When I was saving only one out of these two variants, e2e tests were failing, so I added connection suffix to the tape file names.

I'm setting the priority here as 'high' because I can't get the e2e tests to pass in #1159 without this change.

## Testing Instructions

The e2e tests in CI should pass. 
If you run `pnpm test:playwright:update-tapes`, the tests should pass, and no new tapes should be added.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
